config/placeholder_replacement.go

```diff
@@ -0,0 +1,136 @@
+package config
+
+import (
+	"reflect"
+	"regexp"
+
+	"github.com/pkg/errors"
+	"go.uber.org/multierr"
+	"go.viam.com/rdk/utils"
+)
+
+// Example strings that satisfy this regexp:
+// ${packages.my-COOL-ml-model/__89}
+// ${packages.modules.intel:CameraThatRocks}
+var packagePlaceholderRegexp = regexp.MustCompile(`\$\{(?P<fulltriple>packages(\.(?P<type>ml_models|modules))?\.(?P<name>[\w:\/-]+))\}`)
+
+func ContainsPlaceholder(s string) bool {
+	return packagePlaceholderRegexp.Match([]byte(s))
+}
+
+func (cfg *Config) ReplacePlaceholders() error {
+	var allErrs, err error
+	visitor := NewPlaceholderReplacementVisitor(cfg)
+
+	for i, s := range cfg.Services {
+		cfg.Services[i].Attributes, err = walkTypedAttributes(visitor, s.Attributes)
+		allErrs = multierr.Append(allErrs, err)
+	}
+
+	for i, c := range cfg.Components {
+		cfg.Components[i].Attributes, err = walkTypedAttributes(visitor, c.Attributes)
```

Comment:
Replacing in `ConvertedAttributes` was not getting to the module. I suspect that it is because `ConvertedAttributes` is populated on the module side, not the rdk side

Reply:
Yeah you want to replace in `Attributes` only. `ConvertedAttributes` is derived from `Attributes` and I think we should only replace in the true source.